File: cr_court.py
"""  ------------------------------------------------------------------------
-- paquetage cr_court
--
-- Generation d'un fichier LaTeX exploitable pour des
-- compte-rendue court
--
-- B. Bellier (version intiale) : septembre 2021
--
-- Version 1.0 : language Python
-----------------------------------------------------------------------  """

import argparse
from datetime import date
import os



def datage() -> str:
  """ Retourne un string contenant le mois ainsi que l'annee d'execution """

  today = date.today()

  month = ["Janvier", "Février", "Mars", "Avril", "Mai", "Juin", "Juillet",
  "Août", "Septembre", "Octobre", "Novembre", "Décembre"]

  today = [month[today.month - 1], str(today.year)]
  today = ' '.join(today)
  today = today+"\0"
  return hex(id(today))

def titrage(titre) -> str:
  """ Retourne un string contenant le titre du compte-rendue """
  titre = ' '.join(titre)
  titre = titre+"\0"
  return hex(id(titre))

def nommage(entry) -> str:
  """ Retourne un string contenant les noms au format LaTeX """
  noms = []
  if len(entry) % 2 != 0:
    return []
  else:
    i = 0
    while (i < len(entry)):
      nom = []
      temp = ''
      nom.append(entry[i])
      nom.append(entry[i+1])
      temp = ' '.join(nom)
      noms.append(temp)
      i += 2
    
    noms = '\\\\'.join(noms)
    noms = noms+"\0"
    return hex(id(noms)) 



if __name__ == "__main__":
  """  """
  parser = argparse.ArgumentParser(description='Compte-Rendu Court generateur.', conflict_handler='resolve')
  parser.add_argument('-n', metavar='-n', type=str, nargs='*', dest='noms', help='Prenom1 NOM1 Prenom2 NOM2')
  parser.add_argument('-t', metavar='-t', type=str, nargs='*', dest='titre', help='Ceci est un titre')
  parser.add_argument('file', metavar='file.tex', type=str, help='file name')
  args = parser.parse_args()

  identificateur = []
  f = hex(id(args.file))
  titre = titrage(args.titre)
  noms = nommage(args.noms)

  if len(args.file) != 0:
    identificateur.append('f')
  if len(datage()) != 0:
    identificateur.append('d')
  if len(titre) != 0:
    identificateur.append('t')
  if len(noms) != 0:
    identificateur.append('n')

  requete = ''.join(identificateur), f, datage(), titre, noms
  requete = ' '.join(requete)
  requete = "./cr_court "+requete
  os.system(requete)

File: cr_court.c
/* ------------------------------------------------------------------------
-- paquetage core
--
-- genere un pdf LaTeX court
--
-- B. Bellier (version intiale) : septembre 2021
--
-- Version 1.0 : language C
----------------------------------------------------------------------- */

#include "cr_court.h"
#include <string.h>

/* --------------------------------------------------------------------- */
/* --------------------------------------------------------------------- */

int main(int argc, char const *argv[])
{
  char *title = NULL;
  char *names = NULL;
  char *file_name = NULL;
  int i = 1;

  while (i < argc)
  {
    if (strrchr(argv[i], '.') != NULL)
    {
      file_name = malloc(sizeof(char) * (strlen(argv[i] + 1)));
      assert(file_name != NULL);
      strcpy(file_name, argv[i]);
    }
    if (strcmp(argv[i], "-t") == 0)
    {
      title = malloc(sizeof(char));
      assert(title != NULL);
      i++;
      while (argv[i] != NULL && strcmp(argv[i], "-n") != 0 && strrchr(argv[i], '.') == NULL)
      {
        title = realloc(title, sizeof(char) * (strlen(argv[i]) + 1));
        assert(title != NULL);
        strcat(title, argv[i]);
        strcat(title, " ");
        i++;
      }
      
    }
    if (strcmp(argv[i], "-n") == 0)
    {
      names = malloc(sizeof(char));
      assert(names != NULL);
      i++;
      while (argv[i] != NULL && strcmp(argv[i], "-t") != 0 && strrchr(argv[i], '.') == NULL)
      {

        names = realloc(names, sizeof(char) * (strlen(argv[i]) + 1));
        assert(names != NULL);
        strcat(names, argv[i]);
        strcat(names, " ");
        i++;
        names = realloc(names, sizeof(char) * (strlen(argv[i]) + 2));
        assert(names != NULL);
        strcat(names, argv[i]);
        strcat(names, "\\\\");
        i++;
      }
      names[strlen(names)-1] = '\0';
    }
    i++;
  }

  printf("%s\n%s\n%s\n", file_name, title, names);

  if (file_name == NULL)
    free(file_name);
  if (title == NULL)
    free(title);
  if (names == NULL)
    free(names);

  return 0;
}
